pkg/cloudprovider/aws/suite_test.go

```diff
@@ -165,16 +170,67 @@ var _ = Describe("Allocation", func() {
+						ResourceRequirements: v1.ResourceRequirements{
+							Requests: v1.ResourceList{resources.AWSNeuron: resource.MustParse("1")},
+							Limits:   v1.ResourceList{resources.AWSNeuron: resource.MustParse("1")},
+						},
+					}),
+					test.UnschedulablePod(test.PodOptions{
+						ResourceRequirements: v1.ResourceRequirements{
+							Requests: v1.ResourceList{resources.AWSNeuron: resource.MustParse("1")},
+							Limits:   v1.ResourceList{resources.AWSNeuron: resource.MustParse("1")},
+						},
+					}),
+				)
+				// it should've tried to pack them on a single inf1.6xlarge then hit an insufficient capacity error
+				for _, pod := range pods {
+					ExpectNotScheduled(ctx, env.Client, pod)
 				}
-				Expect(InstancesLaunchedFrom(fakeEC2API.CalledWithCreateFleetInput.Iter())).To(Equal(2))
-				overrides := []*ec2.FleetLaunchTemplateOverridesRequest{}
-				for input := range fakeEC2API.CalledWithCreateFleetInput.Iter() {
-					overrides = append(overrides, input.(*ec2.CreateFleetInput).LaunchTemplateConfigs[0].Overrides...)
+				nodeNames := sets.NewString()
+				for _, pod := range ExpectProvisioned(ctx, env.Client, scheduler, provisioners, provisioner, pods...) {
+					nodeNames.Insert(ExpectScheduledWithInstanceType(ctx, env.Client, pod, "inf1.2xlarge").Name)
 				}
-				for _, override := range overrides {
-					Expect(*override.InstanceType).To(Equal("inf1.6xlarge"))
+				Expect(nodeNames.Len()).To(Equal(2))
+			})
+			It("should launch instances on later recon attempt with Insufficient Capacity Error Cache expiry", func() {
+				fakeEC2API.ShouldTriggerInsufficientCapacity = true
+				pods := ExpectProvisioned(ctx, env.Client, scheduler, provisioners, provisioner,
+					test.UnschedulablePod(test.PodOptions{
+						ResourceRequirements: v1.ResourceRequirements{
+							Requests: v1.ResourceList{resources.AWSNeuron: resource.MustParse("2")},
+							Limits:   v1.ResourceList{resources.AWSNeuron: resource.MustParse("2")},
+						},
+					}),
+					test.UnschedulablePod(test.PodOptions{
+						ResourceRequirements: v1.ResourceRequirements{
+							Requests: v1.ResourceList{resources.AWSNeuron: resource.MustParse("2")},
+							Limits:   v1.ResourceList{resources.AWSNeuron: resource.MustParse("2")},
+						},
+					}),
+				)
+				// it should've tried to pack them on a single inf1.6xlarge then hit an insufficient capacity error
+				for _, pod := range pods {
+					ExpectNotScheduled(ctx, env.Client, pod)
 				}
+				// capacity shortage is over - wait for expiry (N.B. the Karpenter logging will not show the overridden cache expiry in this test context)
+				fakeEC2API.ShouldTriggerInsufficientCapacity = false
+				Eventually(func(g Gomega) int {
+					nodeNames := sets.NewString()
+					for _, pod := range ExpectProvisioned(ctx, env.Client, scheduler, provisioners, provisioner, pods...) {
+						nodeNames = nodeNames.Insert(ExpectScheduledWithInstanceTypeAndGomega(ctx, env.Client, pod, "inf1.6xlarge", g).Name)
```

Comment:
Just preempting questions here - this is a brand new feature of Gomega, which makes this `Eventually` function a lot less complicated than it could've been: https://onsi.github.io/gomega/#eventually